### PR TITLE
A new package: hstr(hh).

### PR DESCRIPTION
**Pull request by weijianwen**

### CODE REVIEW

File: var/spack/repos/builtin/packages/hstr/package.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Hstr(AutotoolsPackage):
+    """hstr(hh) is a shell history suggest box for Bash and Zsh,
+    which enable easy viewing, searching and using
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hstr/package.py]: typo: which _enables_

File: var/spack/repos/builtin/packages/hstr/package.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    homepage = "https://github.com/dvorka/hstr"
+    url      = "https://github.com/dvorka/hstr/archive/1.22.tar.gz"
+
+    version('1.22', '620dab922fadf2858938fbe36d9f99fd')
+
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool',  type='build')
+    depends_on('m4',       type='build')
+    depends_on('ncurses@5.9')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hstr/package.py]: It only works with 5.9?

Comment (weijianwen) [inline var/spack/repos/builtin/packages/hstr/package.py]: Yes, build fails on `ncurses@6.0`. I plan to submit an issue.

Comment (weijianwen) [inline var/spack/repos/builtin/packages/hstr/package.py]: Error messages:

```
In file included from include/hstr_curses.h:25:0,
                 from hstr_curses.c:20:
/lustre/spack/sandybridge/linux-centos7-x86_64/gcc-4.8.5/ncurses-6.0-75hnzrecoib5eqdvfchl6n2jgcxc6dgo/include/unctrl.h:60:38: error: conflicting types for ‘unctrl’
 NCURSES_EXPORT(NCURSES_CONST char *) NCURSES_SP_NAME(unctrl) (SCREEN*, chtype);
                                      ^
In file included from /usr/include/curses.h:1989:0,
                 from include/hstr_curses.h:25,
                 from hstr_curses.c:20:
/lustre/spack/sandybridge/linux-centos7-x86_64/gcc-4.8.5/ncurses-6.0-75hnzrecoib5eqdvfchl6n2jgcxc6dgo/include/unctrl.h:57:38: note: previous declaration of ‘unctrl’ was here
 NCURSES_EXPORT(NCURSES_CONST char *) unctrl (chtype);
                                      ^
make[1]: *** [hstr_curses.o] Error 1
make[1]: *** Waiting for unfinished jobs....
In file included from hstr.c:26:0:
/lustre/spack/sandybridge/linux-centos7-x86_64/gcc-4.8.5/ncurses-6.0-75hnzrecoib5eqdvfchl6n2jgcxc6dgo/include/unctrl.h:60:38: error: conflicting types for ‘unctrl’
 NCURSES_EXPORT(NCURSES_CONST char *) NCURSES_SP_NAME(unctrl) (SCREEN*, chtype);
                                      ^
In file included from /usr/include/curses.h:1989:0,
                 from hstr.c:26:
/lustre/spack/sandybridge/linux-centos7-x86_64/gcc-4.8.5/ncurses-6.0-75hnzrecoib5eqdvfchl6n2jgcxc6dgo/include/unctrl.h:57:38: note: previous declaration of ‘unctrl’ was here
 NCURSES_EXPORT(NCURSES_CONST char *) unctrl (chtype);
                                      ^
mv -f .deps/hstr_regexp.Tpo .deps/hstr_regexp.Po
make[1]: *** [hstr.o] Error 1
mv -f .deps/hstr_blacklist.Tpo .deps/hstr_blacklist.Po
mv -f .deps/hstr_utils.Tpo .deps/hstr_utils.Po
mv -f .deps/hashset.Tpo .deps/hashset.Po
mv -f .deps/hstr_favorites.Tpo .deps/hstr_favorites.Po
mv -f .deps/radixsort.Tpo .deps/radixsort.Po
mv -f .deps/hstr_history.Tpo .deps/hstr_history.Po
make[1]: Leaving directory `/tmp/pytest-of-rpm/pytest-43/test_keep_exceptions0/tmp/spack-stage/spack-stage-d8BDjC/hstr-1.22/src'
make: *** [all-recursive] Error 1
```

Comment (weijianwen) [inline var/spack/repos/builtin/packages/hstr/package.py]: See https://github.com/dvorka/hstr/issues/211 .

**Pull request merged**